packages/@aws-cdk/aws-ecs/lib/base/base-task-definition.ts

```diff
@@ -0,0 +1,227 @@
+import iam = require('@aws-cdk/aws-iam');
+import cdk = require('@aws-cdk/cdk');
+import { ContainerDefinition, ContainerDefinitionProps } from '../container-definition';
+import { cloudformation } from '../ecs.generated';
+
+/**
+ * Basic task definition properties
+ */
+export interface BaseTaskDefinitionProps {
+  /**
+   * Namespace for task definition versions
+   *
+   * @default Automatically generated name
+   */
+  family?: string;
+
+  /**
+   * The IAM role assumed by the ECS agent.
+   *
+   * The role will be used to retrieve container images from ECR and
+   * create CloudWatch log groups.
+   *
+   * @default An execution role will be automatically created if you use ECR images in your task definition
+   */
+  executionRole?: iam.Role;
+
+  /**
+   * The IAM role assumable by your application code running inside the container
+   *
+   * @default A task role is automatically created for you
```

Comment:
I like this default behavior, but how I can intentionally opt out of this (in order to use EC2 instance creds instead if I'm on EC2)?

Reply:
Hmm, task role should not be automatically created since we can't really make assumptions about what permissions this role needs.